catkin_ws/src/state_estimation/src/state_aggregator.py

```diff
@@ -20,18 +20,42 @@
 class State_Aggregator:
 
     def __init__(self):
-        # position
-        self.x = 0.0
-        self.y = 0.0
-        self.z = 0.0
-        self.x_offset = 0.0
-        self.y_offset = 0.0
-
-        # orientation
-        self.q_auv = np.quaternion(1, 0, 0, 0) # w, x, y, z - orientation of AUV as seen from world frame
-        self.q_world = np.quaternion(0, 1, 0, 0) # 'nominal' orientation, relative to global (imu) frame - allows imu reset 
+        # TODO - keep track of position, orientation, as well as pos/q according to each individual sensor
+        # TODO - StateSource class for the state provided by each sensor + 'overall' state of different frames
+
+        '''Overall state'''
+        # overall position (x,y - dvl, z - depth sensor)
+        self.pos_world = np.array([0.0, 0.0, 0.0])
+        self.pos_auv = np.array([0.0, 0.0, 0.0])
+
+        # overall orientation (imu)
+        self.q_global = np.quaternion(0, 1, 0, 0) # orientation of inertial global frame relative to NED (we use NWU convention) - TODO - not needed
+        self.q_world = np.quaternion(1, 0, 0, 0)
+        self.q_auv = np.quaternion(1, 0, 0, 0)
         self.euler = np.array([0.0, 0.0, 0.0]) # to determine when wrap-around happens, tracks q_auv
 
+        '''DVL state'''
+        # position - based on dvl
+        # self.pos_world_dvl = np.array([0.0, 0.0, 0.0]) # x, y, z - 'nominal' position relative to global (dvl) frame
+        self.pos_auv_dvl = np.array([0.0, 0.0, 0.0]) # position of auv relative to world frame (q_world, pos_world_dvl)
+
+        # additionally keep track of orientation from dvl (only for comparison)
+        self.q_dvl_mount = np.quaternion(1, 0, 0, 0) # initial orientation of DVL in AUV frame (measured/assumed based on mounting of sensor) #TODO
+        #TODO - include position mounting for sensors
+        self.q_auv_dvl = np.quaternion(1, 0, 0, 0) # orientation relative to q_world (not q_world_dvl) - to compare with imu
```

Comment:
This variable isn't used but I guess it makes sense to keep track in case we need to use it in the future

Reply:
Yeah, I was thinking of use it to visualize the discrepancy between the orientation according to dvl/imu - maybe also state fusion